src/eos-updater-fetch.c

```diff
@@ -165,6 +248,31 @@ content_fetch (GTask *task,
 
   g_main_context_push_thread_default (task_context);
 
+  /* Do we want to use the new libostree code for P2P, or fall back on the old
+   * eos-updater code?
+   * FIXME: Eventually drop the old code. See:
+   * https://phabricator.endlessm.com/T19606 */
+  if (data->results != NULL)
+    {
+      g_message ("Fetch: using results %p", data->results);
+
+      if (content_fetch_new (updater, data, task_context, cancellable, &error))
+        {
+          g_message ("Fetch: finished pulling using libostree P2P code");
+          g_task_return_boolean (task, TRUE);
+
+          goto cleanup;
```

Comment:
I wonder if instead of using `goto` here, we can split the content-fetch-via-ostree and content-fetch-via-old-code parts into separate functions and dispatch from here.

Reply:
My thinking was to try and minimise the number of changes to the old code path, at least to begin with, to reduce the chance of accidentally breaking it. I might do a follow-up commit to refactor this a bit.

Reply:
Pushed a follow-up commit.